yaml/stage.go

```diff
@@ -23,7 +23,7 @@ type (
 	// nolint:lll // jsonschema will cause long lines
 	Stage struct {
 		Name  string          `yaml:"name,omitempty"  json:"name,omitempty"  jsonschema:"minLength=1,description=Unique identifier for the stage in the pipeline.\nReference: https://go-vela.github.io/docs/concepts/pipeline/stages/"`
-		Needs raw.StringSlice `yaml:"needs,omitempty" json:"needs,omitempty" jsonschema:"description=Stages that must complete before starting the current one.\nReference: https://go-vela.github.io/docs/concepts/pipeline/stages/needs/"`
+		Needs raw.StringSlice `yaml:"needs,omitempty,flow" json:"needs,omitempty" jsonschema:"description=Stages that must complete before starting the current one.\nReference: https://go-vela.github.io/docs/concepts/pipeline/stages/needs/"`
```

Comment:
Updates the `Stage` struct to render the keys with `[]` syntax instead of across multiple lines with `-`. 

The current behavior of this package when utilizing `yaml.Marshal`:

```yaml
needs:
  - clone
```

The updated behavior of this package when utilizing `yaml.Marshal`:

```yaml
needs: [ clone ]
````

 I created a brief example [here](https://play.golang.org/p/ROz6_kjcpFr) to help visualize the difference when utilizing the `flow` flag.